extensions/BMO/web/js/advanced-search.js

```diff
@@ -42,6 +42,10 @@ Bugzilla.BMO.AdvancedSearch = class AdvancedSearch {
 
     // Select non-Graveyard classifications
     for (const $option of this.$classifications.options) {
+      // Return if already selected (part of the default search params for query page)
+      if ($option.selected) {
+        return;
+      }
```

Comment:
Can you elaborate in the commit message how this fixes the issue?

Reply:
Oh, I think I understand: if any `$option` was already selected, then we stop the automatic-setting of `$option.selected` to the default.

However, isn't this order-dependent? Won't this behave incorrectly in the following two scenarios?
1. (Unlikely, I suppose): `Graveyard` is moved to be higher in the list
2. The user _unselects_ `Client Software`, then chooses to remember the search options. This code will start by setting `Client Software` to the default value (enabled), then get to `Developer Infrastructure`, realize that it's been selected due to "search options being remembered", then stop at that point. The end result being: `Client Software` is selected even though it was supposed to be remembered as "unselected".